numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -69,6 +69,14 @@ static NPY_INLINE int PyInt_Check(PyObject *op) {
     #define Npy_EnterRecursiveCall(x) Py_EnterRecursiveCall(x)
 #endif
 
+/* This is a backport of Py_SETREF */
+#define Npy_SETREF(op, op2)                     \
+    do {                                        \
+        PyObject *_py_tmp = (PyObject *)(op);   \
+        (op) = (op2);                           \
+        Py_DECREF(_py_tmp);                     \
+    } while (0)
+
```

Comment:
We should make the backport explicit. Since Py_SETREF was added in 3.5.2 [0], this should have a `#if PY_VERSION_HEX < 0x03050200 ... #else`

[0] see CPython changeset `57a01d3a0ee` and  `git tag --contains 57a01d3a0ee`